internal/cli/server/config.go

```diff
@@ -131,6 +131,10 @@ type P2PConfig struct {
 
 	// Discovery has the p2p discovery related settings
 	Discovery *P2PDiscovery `hcl:"discovery,block" toml:"discovery,block"`
+
+	// TxArrivalWait sets the maximum duration the transaction fetcher will wait for
+	// an announced transaction to arrive before explicitly requesting it
+	TxArrivalWait time.Duration `hcl:"txarrivalwait,optional" toml:"txarrivalwait,optional"`
```

Comment:
I think you will have to add a `Raw string` field here and give the `txarrivalwait` tag to it, and then convert that string to time.Duration.

Have a look at [this](https://github.com/maticnetwork/bor/commit/fcb722e4bf5bdda23d4c91da40ab95c9023a80c3) for reference.

Reply:
I'm not sure I understand the rationale for this.  time.Duration can unmarshal from an hcl and a toml, and there are already multiple other time.Durations in the config structs.  Is there a particular reason why we'd want to use a different unmarshal format for this variable than for the other duration variables in the config?

Reply:
@pratikspatil024 

I added both - thanks for the feedback. 

d56370b

by the way, that pr you linked is something I spend a lot of time thinking about.  we should connect sometime soon and discuss.  That PR opens up a lot of doors that are currently closed and would have a lot of neat secondary effects and enable lots of neat protocols.

Reply:
Sure, let's move the conversation over [there](https://github.com/maticnetwork/bor/commit/fcb722e4bf5bdda23d4c91da40ab95c9023a80c3).